CapstoneProject/Assets/Scripts/Core/Controller/Dialogue/DialogueManager.cs

```diff
@@ -197,33 +213,177 @@ private void BeginText()
     {
         displayTextObject.SetActive(true);
         displayTextObject.GetComponent<TextMeshProUGUI>().enabled = true;
-        StartCoroutine(TextScroll("This is test dialogue. Wa ba ba go bo"));
+        //scrollingCoroutine = TextScrollInput("This is test dialogue.");
+        //StartCoroutine(scrollingCoroutine);
 
+    }
+
+    private IEnumerator MainDialogueLoop()
+    {
+        dialogueIsplaying = true;
+        
+        IEnumerator textCoroutine = null;
         
+        UnityEngine.Debug.Log("Main dialogue loop entered");
+
+
+
+
+        foreach(string dialogue in loadedDialogue)
+        {
+            dialogueTicker++;
+            UnityEngine.Debug.Log("Looping");
+
+            if (gameState.relationshipStatus == 2)
+            {
+                StartCoroutine(SearchForState(textCoroutine, "playerkillrepeat", "break"));
+                dialogueSkipRepeat = true;
+            }
+            else if (gameState.relationshipStatus == 1)
+            {
+                StartCoroutine(SearchForState(textCoroutine, "playerkill", "playerdeathrepeat"));
+                dialogueSkipRepeat = true;
+            }
+            else if (gameState.recentDeath > 2)
+            {
+                StartCoroutine(SearchForState(textCoroutine, "playerkillrepeat", "break"));
+                dialogueSkipRepeat = true;
+            }
+            else if (gameState.recentDeath == 1)
+            {
+                StartCoroutine(SearchForState(textCoroutine, "playerdeathrepeat", "playerkillrepeat"));
+                dialogueSkipRepeat = true;
+            }
+            
+            if (dialogueSkipRepeat)
+            {
+                UnityEngine.Debug.Log("Drawing text");
+                textCoroutine = TextScrollInput(dialogueTransition);
+                dialoguePriority = true;
+                StartCoroutine(textCoroutine);
+                while (dialoguePriority)
+                {
+                    yield return new WaitForSeconds(0.25F);
+                }
+                UnityEngine.Debug.Log("Finished drawing text");
+                StopCoroutine(textCoroutine);
+            }
+
+            if (dialogue == "maindialogue1" && dialogueState != 1)
+            {
+
+            }
+
+            else if (dialogue == "maindialogue1" || dialogue == "maindialogue2" || dialogue == "maindialogue3" || dialogue == "maindialogue4" || dialogue == "playerdeath")
```

Comment:
this seems like something that could be baked into the scriptable object as well as one variable (continues to next or something) if this is to allow it to move to the next dialogue in a series, I would identify what the next dialogue is and use -1 as a flag for not having a next dialogue.